### PR TITLE
Setting components of filterblock, range filter and checkbox filter

### PR DESCRIPTION
**Pull request by nastyakuzusha**

### CODE REVIEW

File: front/src/components/filters/range-filter/RangeFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { RangeFilter } from "../../../services/filterService";
+
+interface RangeFilterProps {
+    filter: RangeFilter;
+}
+
+const RangeFilter = ({filter}: RangeFilterProps) => {
+    console.log(filter);
```
Comment (AbrosimovVladislav) [inline front/src/components/filters/range-filter/RangeFilter.tsx]: remove console log

File: front/src/components/filters/range-filter/RangeFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    console.log(filter.name);
+    return <div>
```
Comment (AbrosimovVladislav) [inline front/src/components/filters/range-filter/RangeFilter.tsx:9]: we can use <></> here

File: front/src/components/filters/checkbox-filter/CheckboxFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { CheckboxFilter } from "../../../services/filterService";
+
+interface CheckboxFilterProps {
+    filter: CheckboxFilter;
+}
+
+const CheckboxFilter = ({filter}: CheckboxFilterProps) => {
+    return <div>
```
Comment (AbrosimovVladislav) [inline front/src/components/filters/checkbox-filter/CheckboxFilter.tsx:9]: we can use <></> here

File: front/src/gc-components/GCCheckbox.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    ml?: number;
+    size?: MantineSize,
+    className?: string;
+    mt?: MantineNumberSize;
+}
+
+export const GCCheckbox = ({ children, ml, className, size, checked, mt, ...props }: GCCheckboxProps) => {
+    return (
+        <Checkbox ml={ml} className={className} mt={mt} {...props}>
+            {children}
```
Comment (AbrosimovVladislav) [inline front/src/gc-components/GCCheckbox.tsx:16]: which child will checkbox have? is it needed here?

File: front/src/services/filterService.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+export enum FilterType {
```
Comment (AbrosimovVladislav) [inline front/src/services/filterService.ts]: data classes should be defined in schema package

File: front/src/services/filterService.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+export interface RangeFilter extends Filter {
+    value: {
+        start: number,
+        end: number,
+    }
+}
+
+export function getAllFilters(): Filter[] {
+    const filters = [{
```
Comment (AbrosimovVladislav) [inline front/src/services/filterService.ts]: plz rename to mockUseGetAllFilters and add case for CheckboxFilter with few values

File: front/src/components/filters/range-filter/RangeFilter.tsx (hunk 2 above)
Comment (nastyakuzusha) [reply front/src/components/filters/range-filter/RangeFilter.tsx:9]: we'll use some classnames here probably, thats why I use div

File: front/src/gc-components/GCCheckbox.tsx (hunk 4 above)
Comment (nastyakuzusha) [reply front/src/gc-components/GCCheckbox.tsx:16]: like text/list?

File: front/src/services/filterService.ts (hunk 5 above)
Comment (nastyakuzusha) [reply front/src/services/filterService.ts]: what is that?

File: front/src/components/filters/range-filter/RangeFilter.tsx (hunk 2 above)
Comment (AbrosimovVladislav) [reply front/src/components/filters/range-filter/RangeFilter.tsx:9]: Ok,np

File: front/src/gc-components/GCCheckbox.tsx (hunk 4 above)
Comment (AbrosimovVladislav) [reply front/src/gc-components/GCCheckbox.tsx:16]: Ok, np

File: front/src/services/filterService.ts (hunk 5 above)
Comment (AbrosimovVladislav) [reply front/src/services/filterService.ts]: image
Like this

File: front/src/services/filterService.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+export function mockUseGetAllFilters(): Filter[] {
+    return [{
+        id: 0,
+        name: "ON SALE",
+        type: FilterType.CHECKBOX,
+        value: "On Sale"
+    } as CheckboxFilter,
+        {
+        id: 0,
```
Comment (AbrosimovVladislav) [inline front/src/services/filterService.ts]: I mean smth other thing
Add filter Brand, with 4 values (Colt,Magpul,Fab-Defence, Daniel Defence)

File: front/src/services/filterService.ts (hunk 5 above)
Comment (nastyakuzusha) [reply front/src/services/filterService.ts]: done

File: front/src/services/filterService.ts (hunk 6 above)
Comment (nastyakuzusha) [reply front/src/services/filterService.ts]: done

File: front/src/services/filterService.ts (hunk 7 above)
Comment (nastyakuzusha) [reply front/src/services/filterService.ts]: ok

Comment (nastyakuzusha) [reply front/src/services/filterService.ts]: done

**Review by AbrosimovVladislav — APPROVED** [review]

**Pull request merged**